src/plugins/telemetry/server/plugin.ts

```diff
@@ -134,7 +134,12 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
     return {
       getIsOptedIn: async () => {
         const internalRepository = new SavedObjectsClient(savedObjectsInternalRepository);
-        const telemetrySavedObject = await getTelemetrySavedObject(internalRepository);
+        let telemetrySavedObject: TelemetrySavedObject = false; // if an error occurs while fetching opt-in status, a `false` result indicates that Kibana cannot opt-in
+        try {
+          telemetrySavedObject = await getTelemetrySavedObject(internalRepository);
+        } catch (err) {
+          this.logger.debug('Failed to check telemetry opt-in status: ' + err.message);
+        }
```

Comment:
It feels like we should move this to `getTelemetrySavedObject`, as it already has a try/catch to handle a few error cases anyway.

Basically updating this to swallow (& log) errors, and just return `false` if there isn't a `NotFoundError`:
https://github.com/elastic/kibana/blob/c2641bd8cc875dd50c63a40fbaf3054780cd570b/src/plugins/telemetry/server/telemetry_repository/get_telemetry_saved_object.ts#L23-L33

(Then we could also update the unit tests to reflect the expected behavior).

Reply:
I started with that, but I don't have access to a `Logger` instance there, and there are several other consumers of `getTelemetrySavedObject` that also don't have a `Logger` already. So I figured I'd do this instead, since it's only a few lines of code in one file.

We probably don't _need_ to log a message when this happens anyway, so I could change it there and just omit the log. LMK what you'd prefer.

Reply:
> I don't have access to a `Logger` instance there, and there are several other consumers of `getTelemetrySavedObject` that also don't have a `Logger`

Ah, makes sense. Yeah AFAICT there are two places this is used that don't already have access to a logger. One of them is just swallowing errors completely anyway: 
https://github.com/elastic/kibana/blob/c2641bd8cc875dd50c63a40fbaf3054780cd570b/src/plugins/telemetry/server/collectors/telemetry_plugin/telemetry_plugin_collector.ts#L45-L52

And the other is a route that we could fairly easily add a logger to: https://github.com/elastic/kibana/blob/c2641bd8cc875dd50c63a40fbaf3054780cd570b/src/plugins/telemetry/server/routes/telemetry_user_has_seen_notice.ts#L25-L27

The rest of the usages are still internal to the telemetry plugin, so I think it would be worth considering centralizing things inside `getTelemetrySavedObject`, even if it means passing a required (or optional?) `Logger`.

But before spending time on it, let's wait and see what @afharo and @Bamieh think about this whole thing.

Reply:
On another note, I noticed we have the same logic in `telemetry/public/plugin.ts`, so wherever we end up on this, perhaps we should update in both places for consistency: https://github.com/elastic/kibana/blob/c2641bd8cc875dd50c63a40fbaf3054780cd570b/src/plugins/telemetry/public/plugin.ts#L287-L311

Reply:
Thank you for highlighting the places where this is used without catching

I think that this one probably should not catch at all. It's a collector, and collectors are caught at the top level. Catching this may lead to wrong data. What do you think @Bamieh?
https://github.com/elastic/kibana/blob/c2641bd8cc875dd50c63a40fbaf3054780cd570b/src/plugins/telemetry/server/collectors/telemetry_plugin/telemetry_plugin_collector.ts#L45-L52

The route should fail if there is no ES connection. That route is all about updating the SO, so if we catch the GET operation, it will fail on the PUT anyways
https://github.com/elastic/kibana/blob/c2641bd8cc875dd50c63a40fbaf3054780cd570b/src/plugins/telemetry/server/routes/telemetry_user_has_seen_notice.ts#L25-L27

On the public one, I think we're OK with not catching. If there's an ES connection issue, the UI will probably be unusable for many other reasons. This one will only log the error in the browser's console.
https://github.com/elastic/kibana/blob/c2641bd8cc875dd50c63a40fbaf3054780cd570b/src/plugins/telemetry/public/plugin.ts#L287-L311